packages/shared/src/components/cards/ReferralWidget.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { ReactElement } from 'react';
+import { Button, ButtonSize } from '../buttons/Button';
+import {
+  getFacebookShareLink,
+  getTwitterShareLink,
+  getWhatsappShareLink,
+} from '../../lib/share';
+import WhatsappIcon from '../icons/Whatsapp';
+import FacebookIcon from '../icons/Facebook';
+import TwitterIcon from '../icons/Twitter';
+import { TextField } from '../fields/TextField';
+import { ReferralCampaignKey, useReferralCampaign } from '../../hooks';
+import { link } from '../../lib/links';
+import { useShareOrCopyLink } from '../../hooks/useShareOrCopyLink';
+import { labels } from '../../lib';
+import { AnalyticsEvent, TargetId } from '../../lib/analytics';
+import { SimpleTooltip } from '../tooltips';
+
+const ReferralWidget = (): ReactElement => {
+  const { isReady, url } = useReferralCampaign({
```

Comment:
Based on this in the DR:

> to get the right URL we send a query which won’t be at the same time we fetch the user information, so rendering the user details might come in first then the campaign URL which would produce the layout shift, make sure to wait for the campaign query to finish before displaying the content.

We want to display user profile after the url is fetched, but also fetch the url after the profile query has finished so we can validate if the user is seeing their own profile.

I think having these both would place us in a loop and would require us to add this rule again:
```
// @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
// eslint-disable-next-line react-hooks/rules-of-hooks
```
since we would want to render the hook to get the referral url conditionally after the profile query has finished, and I don't know if we would want to introduce more of this...
I'm thinking more of we render the profile and the referral widget, just render that disabled until the url is fetched...which would not introduce a layout shift?

Would like to hear your thoughts on this, can definitely quickly change to whatever we think is best.

Reply:
I think you just need to use `useReferralCampaign`  also in the `ProfileLayout` and once it `isReady` you render its content.

Reply:
That wouldn't be enough as it would mean we would be fetching the referral url even if the user profile wasn't their own. Meaning to use the referral hook we would need to wait until user profile info is fetched and compared if it is their own, if so fetch the referral link and then once the referral link is ready show the profile. 
@sshanzel suggested I use `enabled` option in useQuery for referral invite, so am looking at that opportunity now.

Reply:
There are two things here I believe:
- The first is identifying whether the user is viewing his own profile page which we will pass the value if the query should be enabled or not.
- Second thing is, to avoid the layout shift, for that, we will wait if the referral campaign query to be finished which we can utilize the `isReady` to see if the query is finished.

Reply:
Yup, the second is quite straightforward, the first one is the complex bit, as the `enabled` should change after the profile info is fetched

Reply:
Also, just be wary with regards to the `isReady` evaluation, our current `react-query`, I believe would return `isSuccess: false` if it is `enabled: false` as opposed to the latest version which would return `isSuccess: true`.

Reply:
I have a question though, are we really against showing profile info if the url is not ready? Any reasons behind this except layout shift? 
Personally I believe making a whole profile wait for the referral url is too much, what if there is sth wrong with the api for referral url, will we never show the profile? I'm quite pro to not delaying profile until referral url is there, but rather we just manage the design to not introduce layout shift (things like make copy button disabled until ready, have a placeholder widget until url ready). I might be missing context, but would be curious to know.

Reply:
Layout shift is bad UX, and part of the reason our design team is always going for the extra mile to build the skeletons is to avoid it.

I agree that the page can get held up due to some failing services and we also have done what you suggested, in the past, where we set a specific height to some invisible element, though that is only for elements available every time for everyone. This is only available to you when viewing your own profile, so I don't think the impact would be high in terms of the slight delay.

Reply:
Would love to hear others' thoughts too. cc @dailydotdev/web-team

Reply:
> where we set a specific height to some invisible element

Similar though I'm not suggesting this, I'm suggesting either skeleton for the widget or a disabled state to the copy button, or a loading state.. Sth like this, and this would be for sure only for users who are viewing their own profile

Reply:
For now let's not slow down the development and keep it as blocking.
Feel free to open a separate discussion for improvement with design, in the end it's a UX issue and they need to advise.

Reply:
I'll move this conversation at referral channel then